### PR TITLE
added PhyssymXmlToOsmConverter

### PR DESCRIPTION
**Pull request by carloscaldas**
Closes https://github.com/LBNL-UCB-STI/beam/issues/3076

This change is [Reviewable](https://reviewable.io/reviews/lbnl-ucb-sti/beam/3090)

image

### CODE REVIEW

**Comment by carloscaldas** [comment]
how to test it?
* tool (matsimNetwork1) -> OSM-network1
* rumBeam(OSM-network) -> matsim-network2
* tool (matsimNetwork2) -> OSM-network2
* OSM-network1 == OSM-network2  <-- ?

File: src/test/scala/beam/router/MatsimNetworkToOsmConverter.scala [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  private val changeset = "1"
+    val xmlEncode = "UTF-8"
+
+  lazy val matsimNetwork: Network = {
+    val networkCoordinator = DefaultNetworkCoordinator(beamConfig)
+    networkCoordinator.init()
+    networkCoordinator.network
+  }
+
+  def buildXmlFile(outputFile: Path): Path = {
```
Comment (JustinPihony) [inline src/test/scala/beam/router/MatsimNetworkToOsmConverter.scala:28]: Should everything except this be private? Also, should there be a documented usecase and/or test to show usage better?

**Review by JustinPihony — APPROVED** [review]
Some minor comments that should still be reviewed - as well as you may want somebody who is closer to the XML itself to review.

File: src/test/scala/beam/router/MatsimNetworkToOsmConverter.scala (hunk 1 above)
Comment (carloscaldas) [reply src/test/scala/beam/router/MatsimNetworkToOsmConverter.scala:28]: I agree. I am adding your suggestions

Comment (carloscaldas) [reply src/test/scala/beam/router/MatsimNetworkToOsmConverter.scala]: this file was removed. This task is to convert from `Matsim physsym XML` to `OSM XML`

File: src/test/scala/beam/router/PhyssymXmlToOsmConverter.scala [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+
+        xmlStreamWriter.writeAttribute("lat", node.wgsCoordinate.latitude.toString)
+        xmlStreamWriter.writeAttribute("lon", node.wgsCoordinate.longitude.toString)
+
+        xmlStreamWriter.writeCharacters(breakLine)
+      }
+    }
+  }
+
+  def main(args: Array[String]): Unit = {
```
Comment (REASY) [inline src/test/scala/beam/router/PhyssymXmlToOsmConverter.scala:261]: nitpick: an example of run so it is easy to use

**Review by REASY — CHANGES_REQUESTED** [review]

**Comment by crixal** [comment]
How do you plan to use the converter? Run it and use osmium tool to convert xml to osm.pbf?

**Comment by crixal** [comment]
You can create Osm file at once for example https://github.com/LBNL-UCB-STI/beam/blob/f50e73d9a13c6dd43164ad676c6c64503b6e558f/src/main/scala/beam/router/cch/CchWrapper.scala#L54

File: src/test/scala/beam/router/PhyssymXmlToOsmConverter.scala (hunk 2 above)
Comment (carloscaldas) [reply src/test/scala/beam/router/PhyssymXmlToOsmConverter.scala:263]: changes done and waiting for your review

**Review by REASY — APPROVED** [review]

**Pull request merged**